files-csharp/FastestWayToReadATextFileInCsharp/FastestWayToReadATextFileInCsharp/WaysToReadATextFileInCsharp.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System.Text;
+
+namespace FastestWayToReadATextFileInCsharp
+{
+    public class WaysToReadATextFileInCsharp
+    {
+        private readonly string _sampleFilePath;
+
+        public WaysToReadATextFileInCsharp(string sampleFilePath)
+        {
+            _sampleFilePath = sampleFilePath;
+        }
+
+        public string UseFileReadAllLines()
+        {
+            var stringArray = File.ReadAllLines(_sampleFilePath);
+
+            return string.Join(Environment.NewLine, stringArray);
+        }
+
+        public string UseFileReadAllText() => File.ReadAllText(_sampleFilePath);
+        
+        public string UseFileReadLines()
+        {
+            var stringArray = File.ReadLines(_sampleFilePath);
+
+            return string.Join(Environment.NewLine, stringArray);
+        }
+
+        public string UseStreamReaderReadLine()
+        {
+            using var streamReader = new StreamReader(_sampleFilePath);
+            var stringBuilder = new StringBuilder();
+            string fileLine;
+
+            while ((fileLine = streamReader.ReadLine()) != null)
+            {
+                stringBuilder.AppendLine(fileLine);
+            }
+
+            return stringBuilder.ToString().TrimEnd();
```

Comment:
Can this TrimEnd be avoided? This will cause a second string allocation before the return.

Reply:
@jshergal, To fix the TrimEnd issue, we can fix use Append and append the NewLine character manually after each line, this will give us more control on when to put the NewLine and we can avoid that extra NewLine that AppendLine does at the end of the file and we subsequently avoid the TrimEnd(); 
Do you see any flaws in this approach? apart from the potential end of line fidelity loss?

Reply:
@karimkod - That approach would work, with the caveat of the potential loss of fidelity. I had posed the question as I didn't run the code to check exactly what we were trimming. Having looked at it a few times now, I am confident that we were doing this to remove the final newline that was added by the AppendLine. With that being the case, we could also take the approach of removing it from the StringBuilder buffer before calling to string. Something like this (I didn't test this and it might have an off by one error as I am just freehanding this - also maybe it could be simplified a little)
`stringBuilder.Remove(stringBuilder.Length - Environment.NewLine.Length - 1, Environment.NewLine.Length);`